packages/cli/lib/watch.js

```diff
@@ -0,0 +1,255 @@
+    const relativePath = file.replace(regex, '');
+    return convertToUnixPath(path.join(dest, relativePath));
+  };
+
+  if (!disableInitial) {
+    // Use uploadFolder so that failures of initial upload are retried
+    uploadFolder(accountId, src, dest, { mode }, commandOptions, filePaths)
+      .then(result => {
+        logger.success(
+          `Completed uploading files in ${src} to ${dest} in ${accountId}`
+        );
+        if (postInitialUploadCallback) {
+          postInitialUploadCallback(result);
+        }
+      })
+      .catch(error => {
+        logger.error(
+          `Initial uploading of folder "${src}" to "${dest} in account ${accountId} failed`
+        );
+        logErrorInstance(error, {
+          accountId,
+        });
+      });
+  }
+
+  watcher.on('ready', () => {
+    logger.log(
+      `Watcher is ready and watching ${src}. Any changes detected will be automatically uploaded and overwrite the current version in the developer file system.`
+    );
+  });
+
+  watcher.on('add', async filePath => {
+    const destPath = getDesignManagerPath(filePath);
+    const uploadPromise = uploadFile(accountId, filePath, destPath, {
+      src,
+      mode,
+      commandOptions,
+    });
+    triggerNotify(notify, 'Added', filePath, uploadPromise);
+  });
+
+  if (remove) {
+    const deleteFileOrFolder = type => filePath => {
+      // If it's a fields.js file that is in a module folder or the root, then ignore because it will not exist on the server.
+      if (isConvertableFieldJs(src, filePath, commandOptions.convertFields)) {
+        return;
+      }
+
+      const remotePath = getDesignManagerPath(filePath);
+      if (shouldIgnoreFile(filePath)) {
+        logger.debug(`Skipping ${filePath} due to an ignore rule`);
+        return;
+      }
+
+      logger.debug('Attempting to delete %s "%s"', type, remotePath);
+      queue.add(() => {
+        const deletePromise = deleteRemoteFile(accountId, filePath, remotePath)
+          .then(() => {
+            logger.log('Deleted %s "%s"', type, remotePath);
+          })
+          .catch(error => {
+            logger.error('Deleting %s "%s" failed', type, remotePath);
+            logApiErrorInstance(
+              error,
+              new ApiErrorContext({
+                accountId,
+                request: remotePath,
+              })
+            );
+          });
+        triggerNotify(notify, 'Removed', filePath, deletePromise);
+        return deletePromise;
+      });
+    };
+
+    watcher.on('unlink', deleteFileOrFolder('file'));
+    watcher.on('unlinkDir', deleteFileOrFolder('folder'));
+  }
+
+  watcher.on('change', async filePath => {
+    const destPath = getDesignManagerPath(filePath);
+    const uploadPromise = uploadFile(accountId, filePath, destPath, {
+      src,
+      mode,
+      commandOptions,
+    });
+    triggerNotify(notify, 'Changed', filePath, uploadPromise);
+  });
+
+  return watcher;
+}
+
+module.exports = {
+  watch,
+};
```

Comment:
@kemmerle We actually decided to keep a slightly refactored version of watch in `local-dev-lib` since it's used by the vscode extension (see https://github.com/HubSpot/hubspot-local-dev-lib/pull/28). My bad for not updating the issue!

Reply:
Addressed in https://github.com/HubSpot/hubspot-cli/pull/934/commits/d5cd90f9d4931b2d378f9e1f3e6b6e312bcf5e25